src/Microsoft.Extensions.Primitives/Extensions.cs

```diff
@@ -14,6 +14,17 @@ public static class Extensions
         /// <param name="segment">The <see cref="StringSegment"/> to add.</param>
         /// <returns>The original <see cref="StringBuilder"/>.</returns>
         public static StringBuilder Append(this StringBuilder builder, StringSegment segment)
+        {
+            return builder.Append(in segment);
+        }
+
+        /// <summary>
+        /// Add the given <see cref="StringSegment"/> to the <see cref="StringBuilder"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="StringBuilder"/> to add to.</param>
+        /// <param name="segment">The <see cref="StringSegment"/> to add.</param>
+        /// <returns>The original <see cref="StringBuilder"/>.</returns>
+        public static StringBuilder Append(this StringBuilder builder, in StringSegment segment)
```

Comment:
I believe this is a source breaking change until 7.3.

Anyone that was previously calling the `public static StringBuilder Append(this StringBuilder builder, StringSegment segment)` overload will now get a `CS0121: The call is ambiguous between the following methods or properties`.

Reply:
There is also some weirdness in that the v7.1 version of the compiler will allow the above, as it will see the new overload as `ref` and allow the compiler to call it and disambiguate with `ref`, but that will break if recompiled using v7.2+ since `ref` cannot be used for `in` parameters (I'm going to log a bug for this in a minute).

Reply:
:-/

Reply:
https://github.com/dotnet/csharplang/issues/1308